pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -370,6 +379,55 @@ func (b *EndpointBuilder) BuildClusterLoadAssignment(endpointIndex *model.Endpoi
 	return l
 }
 
+func (b *EndpointBuilder) generateToServiceWaypoint() []*LocalityEndpoints {
+	if b.proxy.IsWaypointProxy() || b.dir != model.TrafficDirectionOutbound || !features.EnableHBONE {
+		return nil
+	}
+	waypoint := b.toServiceWaypoint()
+	if waypoint == "" {
+		return nil
+	}
+
+	serviceVIPs := b.service.ClusterVIPs.GetAddressesFor(b.clusterID)
+	if len(serviceVIPs) == 0 {
+		// TODO log warn
+		return nil
+	}
+	serviceVIP := serviceVIPs[0]
+
+	ep := &endpoint.LbEndpoint{
+		HealthStatus: corev3.HealthStatus_UNKNOWN,
+		LoadBalancingWeight: &wrapperspb.UInt32Value{
+			Value: 1,
+		},
+		HostIdentifier: &endpoint.LbEndpoint_Endpoint{
+			Endpoint: &endpoint.Endpoint{
+				Address: util.BuildInternalAddressWithIdentifier(connectOriginate, net.JoinHostPort(serviceVIP, strconv.Itoa(b.port))),
```

Comment:
1. For multi-cluster we really need to switch to using hostnames. There are a lot of TODOs throughout the ambient code here and in zTunnel related to handling hostnames. 
2. The code at the time of this comment is sending to the waypoint VIP (that's what `waypoint` is when we pass to` BuildTunnelMetadataStruct`). I'm going to replace this with something that expands to waypoint instances, load balances and consults the `TunnelHTTP` option.

Reply:
Maybe rename the variable - 'serviceVIP' seems to suggest it's the service VIP, not the waypoint VIP. And the code above seemed to match that. 

It doesn't have to expand to waypoint instances - ztunnel can probably route to the right instance, and the waypoint may as well be external. Quite the opposite - using waypoint instance is far more risky, since the 'secure naming' auth in ztunnel for instance is far more dangerous and insecure for instances ( Pod IP reuse ) than services VIPs. 

Regarding multi-cluster - I agree on what we need - my point is to have clear coments/docs on the limitations. I know ambient is currently targeted to single-cluster - but sidecars and Istio may be used with MC. Maybe add a check and ignore remote services ?